src/group_theory/free_group.lean

```diff
@@ -863,4 +884,64 @@ fintype.subtype (list.to_finset $ red.enum L₁) $
 
 end reduce
 
+section metric
+
+variable [decidable_eq α]
+
+/-- The length of reduced words provides a norm on a free group. --/
+def norm (x : free_group α) : nat := x.to_word.length
+
+lemma norm_inv_le (x : free_group α) : norm x⁻¹ ≤ norm x :=
+begin
+  let w := x.to_word,
+ 
+  calc norm x⁻¹ = norm ((mk (x.to_word))⁻¹) : by simp only [to_word.mk]
+       ... = norm (mk (inv_rev w)) : by simp only [←inv_mk]
+       ... ≤ (inv_rev w).length : by exact to_word.le_length
+       ... = w.length : inv_rev_length
+       ... = norm x : rfl
+end
+
+lemma norm_inv_eq (x : free_group α) : norm x = norm x⁻¹ :=
+begin
+  refine le_antisymm _ (norm_inv_le x),
+  simpa using norm_inv_le x⁻¹
+end
+
+lemma norm_zero_eq_one (x : free_group α ) : norm x = 0 → x = 1 := 
+begin
+  intro h1,
+  unfold norm at h1,
+  rw list.length_eq_zero at h1,
+  
+  have h2 : (1 : free_group α).to_word = list.nil,
+  { rw one_eq_mk,
+    unfold to_word,
+    simp only [quot_lift_mk],
+    rw ← free_group.red.nil_iff ,
+    simp only [reduce.red], },
+
+  rw ← h2 at h1,
+  exact to_word.inj x 1 h1,
+end
+
+lemma norm_triangle (x : free_group α) (y : free_group α) : norm (x * y) ≤ norm x + norm y :=
```

Comment:
It is called `norm_add_le` for normed groups. Probably, should be called `norm_mul_le` in this case.

Reply:
That name change is made in 64bce9a5329c39b5a527f10bcccd6eee50438246.